armi/reactor/converters/blockConverters.py

```diff
@@ -547,6 +546,13 @@ def convert(self):
             self._buildNthRing(pinComponents, ring)
         self._buildNonPinRings(nonPins)
         self._addDriverFuelRings()
+
+        for comp in self.convertedBlock.getComponents():
+            assert comp.getArea() >= 0.0, (
+                f"{comp} in {self.convertedBlock} has a negative area of {comp.getArea()}. "
+                "Negative areas are not supported."
+            )
```

Comment:
This is a tricky one because this assertion happens in the code on run time to enforce that there are no negative area components. Up above it checks for negative area components and strips them out. I can try to add a test.